Algorithms/Search/BinarySearcher.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+
+namespace Algorithms.Searches
+{
+    public class BinarySearcher<T> where T : IComparable<T>
+    {
+        /// <summary>
+        /// Finds index of first item in array that equals to item to search
+        /// Time complexity: O(logN)
+        /// Space complexity: O(1)
+        /// </summary>
+        /// <param name="sortedData">Sorted array to search in</param>
+        /// <param name="item">Item to search</param>
+        /// <returns>Index of first item that equals to item to search or -1 if none found</returns>
+        public int FindIndex(T[] sortedData, T item) 
+        {
+            var startIndex = 0;
+            var finishIndex = sortedData.Length - 1;
```

Comment:
The algorithm doesn't go through all items, so those are not start and end indices, they are left and right.